src/FINDOLOGIC/Api/RequestBuilders/RequestBuilder.php

```diff
@@ -0,0 +1,238 @@
+        return $this;
+    }
+
+    /**
+     * Sets the query param. It is used as the search query.
+     *
+     * @param $value string
+     * @see https://docs.findologic.com/doku.php?id=integration_documentation:request#search_parameter
+     * @return $this
+     */
+    public function setQuery($value)
+    {
+        $validator = new ParameterValidator([QueryParameter::QUERY => $value]);
+        $validator->rule('string', QueryParameter::QUERY);
+
+        if (!$validator->validate()) {
+            throw new InvalidParamException(QueryParameter::QUERY);
+        }
+
+        $this->addParam(QueryParameter::QUERY, $value);
+        return $this;
+    }
+
+    /**
+     * Sets the count param. It is used to set the number of products that should be displayed.
+     *
+     * @see https://docs.findologic.com/doku.php?id=integration_documentation:request#limiting_paging_parameters
+     * @param $value int
+     * @return $this
+     */
+    public function setCount($value)
+    {
+        $validator = new ParameterValidator([QueryParameter::COUNT => $value]);
+        $validator->rule('equalOrHigherThanZero', QueryParameter::COUNT);
+
+        if (!$validator->validate()) {
+            throw new InvalidParamException(QueryParameter::COUNT);
+        }
+
+        $this->addParam(QueryParameter::COUNT, $value);
+        return $this;
+    }
+
+    /**
+     * Adds the group param. It is used to show only products for one or more specific groups.
+     *
+     * @see https://docs.findologic.com/doku.php?id=integration_documentation:request#limiting_paging_parameters
+     * @see https://docs.findologic.com/doku.php?id=smart_suggest_new#request
+     * @param $value string
+     * @return $this
+     */
+    public function addGroup($value)
+    {
+        $validator = new ParameterValidator([QueryParameter::GROUP => $value]);
+        $validator->rule('string', QueryParameter::GROUP);
+
+        if (!$validator->validate()) {
+            throw new InvalidParamException(QueryParameter::GROUP);
+        }
+
+        $this->addParam(QueryParameter::GROUP, ['' => $value], self::ADD_VALUE);
+        return $this;
+    }
+
+    /**
+     * Adds an own param to the parameter list. This can be useful if you want to put some special key value pairs to
+     * get a different response from FINDOLOGIC.
+     *
+     * Important Note: Both $key and $value are NOT validated. **Using this is neither recommended nor
+     * supported.** If you think any parameter is missing for doing FINDOLOGIC requests, please create an issue before
+     * using this method.
+     *
+     * @param $key string
+     * @param $value mixed
+     * @param $method string Use ParameterBuilder::ADD_VALUE to add the param and not overwrite existing ones and
+     * ParameterBuilder::SET_VALUE to overwrite existing params.
+     *
+     * @return $this
+     */
+    public function addIndividualParam($key, $value, $method)
```

Comment:
I could but I do not want to expose this function, since this is a method I only want to have available for internal usage. 
Also right above in the PHPDoc I have a comment where I advise against using this, since it only should be used if there is **really no other way**.